kyo-cats/shared/src/main/scala/kyo/Cats.scala

```diff
@@ -16,31 +15,35 @@ object Cats:
       * @return
       *   A Kyo effect that, when run, will execute the cats.effect.IO
       */
-    def get[A](io: CatsIO[A])(using Frame): A < Cats =
-        ArrowEffect.suspendMap(Tag[GetCatsIO], io)(Abort.get(_))
+    def get[A](io: CatsIO[A])(using Frame): A < (Abort[Throwable] & Async) =
+        IO {
+            import cats.effect.unsafe.implicits.global
+            val p                = new IOPromise[Throwable, A]
+            val (result, cancel) = io.unsafeToFutureCancelable()
+            result.onComplete(t => p.complete(Result.fromTry(t)))(ExecutionContext.parasitic)
+            p.onInterrupt(_ => discard(cancel()))
```

Comment:
Do we need to await `cancel`? It returns a `Future[Unit]`?

Reply:
Just evaluating the thunk is enough to dispatch the interrupt. Kyo currently doesn't implement backpressure for interrupts, unlike other effect systems. I'm not convinced it's a good behavior and it's also not the case in the Twitter Future. When a system is under pressure, it's best to offload internal workload as soon as possible and ensure there's proper backpressure on the source of incoming workload via a solution like Kyo's scheduler admission control.